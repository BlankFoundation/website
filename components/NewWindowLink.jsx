const NewWindowLink = (props) => {
  return (
     <a {...props} target="_blank" rel="noopener noreferrer">
       {props.children}
    </a>
  );
}

NewWindowLink.defaultProps = {
  
  href: "https://www.example.com",
  children: "Example Link",
  title: "Blank.Foundation",
  social: [{"href":"https://discord.gg/HZM2jcHdEU","name":"Discord"},{"href":"https://twitter.com/blankfdn","name":"Twitter"},{"href":"https://medium.com/@blankfdn","name":"Medium"},{"href":"https://www.notion.so/blankfoundation/Blank-Home-f3fd5330df0d4933aa41e363719c2625","name":"Notion"}]
}

export default NewWindowLink;