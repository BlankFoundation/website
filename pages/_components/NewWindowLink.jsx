const NewWindowLink = (props) => {
  return (
     <a {...props} target="_blank" rel="noopener noreferrer">
       {props.children}
    </a>
  );
}

NewWindowLink.defaultProps = {
  href: "https://www.example.com",
  children: "Example Link"
}

export default NewWindowLink;