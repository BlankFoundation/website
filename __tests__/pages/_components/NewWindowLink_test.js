import React from 'react';
import TestRenderer from 'react-test-renderer';
import NewWindowLink from '../../../pages/_components/NewWindowLink';

describe("NewWindowLink", () => {
  it('renders correctly', () => {
    const rendered = TestRenderer.create(<NewWindowLink />).toJSON();
    expect(rendered).toMatchSnapshot()
  });
});