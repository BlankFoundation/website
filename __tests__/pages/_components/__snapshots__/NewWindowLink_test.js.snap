// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewWindowLink renders correctly 1`] = `
<a
  props={
    Object {
      "children": "Example Link",
      "href": "https://www.example.com",
      "social": Array [
        Object {
          "href": "https://discord.gg/HZM2jcHdEU",
          "name": "Discord",
        },
        Object {
          "href": "https://twitter.com/blankfdn",
          "name": "Twitter",
        },
        Object {
          "href": "https://medium.com/@blankfdn",
          "name": "Medium",
        },
      ],
      "title": "Blank.Foundation",
    }
  }
  rel="noopener noreferrer"
  target="_blank"
/>
`;
